olive/passes/pytorch/genai_model_exporter.py

```diff
@@ -0,0 +1,91 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+# Export a PyTorch model using the onnxruntime-genai package.
+# --------------------------------------------------------------------------
+import logging
+from enum import Enum
+from pathlib import Path
+from typing import Any, Callable, Dict
+
+from onnx import TensorProto
+
+from olive.common.pydantic_v1 import validator
+from olive.hardware.accelerator import AcceleratorLookup, AcceleratorSpec, Device
+from olive.model import ONNXModelHandler, PyTorchModelHandler
+from olive.model.utils import resolve_onnx_path
+from olive.passes import Pass
+from olive.passes.olive_pass import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class GenAIModelExporter(Pass):
+    class Precision(str, Enum):
+        FP32 = "fp32"
+        FP16 = "fp16"
+        INT4 = "int4"
+
+    @staticmethod
+    def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        return {
+            "precision": PassConfigParam(
+                type_=str,
+                required=True,
+                description="Precision of model.",
+            )
+        }
+
+    @staticmethod
+    def _validate_precision(v):
+        valid_precisions = [d.value for d in GenAIModelExporter.Precision]
+        if v not in valid_precisions:
+            raise ValueError(f"Invalid precision: {v}. Valid values are: {valid_precisions}")
+        return v
+
+    @staticmethod
+    def _validators() -> Dict[str, Callable]:
+        return {
+            "validate_precision": validator("precision", allow_reuse=True)(GenAIModelExporter._validate_precision),
+        }
+
+    def _run_for_config(
+        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> ONNXModelHandler:
+        from argparse import Namespace
+
+        from olive.passes.pytorch.genai_exporter import create_model
```

Comment:
The `genai_exporter` module/file is missing. Will this be added?

Reply:
This file is meant to be part of the GenAI package but the project isn't setup for installation so I had to temporarily pull in a local copy for testing. This is only a draft PR and cannot be merged in just yet. The GenAI package has to be setup correctly (API is still in flux) and needs to be published.

Reply:
Here's the source file location - https://github.com/microsoft/onnxruntime-genai/blob/main/src/python/models/export.py